app/models/spree/credit_card_decorator.rb

```diff
@@ -25,4 +28,16 @@ def has_payment_profile? # rubocop:disable Style/PredicateName
   def save_requested_by_customer?
     !!@save_requested_by_customer
   end
+
+  private
+
+  def default_missing?
+    user.credit_cards.where(is_default: true).none?
```

Comment:
⚠️ watch out! this will trigger a DB query that will return the whole credit card data from DB, to later check just if it exists with Enumerable's `none?`. So all the data processing and networking cost will be in vain. Whereas, if you use [exists?](https://apidock.com/rails/ActiveRecord/Base/exists%3f/class) you'll get only the boolean you care about here.

See both queries

**none?**

```ruby
irb(main):004:0> user.credit_cards.where(is_default: true).to_sql
=> "SELECT \"spree_credit_cards\".*
FROM \"spree_credit_cards\" 
WHERE \"spree_credit_cards\".\"user_id\" = 1
  AND \"spree_credit_cards\".\"is_default\" = 't'"
```

**exists?**

```ruby
irb(main):005:0> user.credit_cards.exists?(is_default: true).to_sql
SELECT 1 AS one
FROM "spree_credit_cards"
WHERE "spree_credit_cards"."user_id" = 1
  AND "spree_credit_cards"."is_default" = 't' LIMIT 1
```

Reply:
Nice pickup, thanks! Should have actually looked at the ActiveRecord API...

Reply:
Done